llvm/utils/TableGen/SubtargetEmitter.cpp

```diff
@@ -357,8 +367,10 @@ unsigned SubtargetEmitter::cpuKeyValues(raw_ostream &OS,
   for (const Record *Processor : ProcessorList) {
     StringRef Name = Processor->getValueAsString("Name");
     ConstRecVec FeatureList = Processor->getValueAsListOfDefs("Features");
+    checkDuplicateCPUFeatures(Name, "Features", FeatureList);
```

Comment:
Can the same feature appear in both Features and TuneFeatures? If not, should we check that as well?

Reply:
> Can the same feature appear in both Features and TuneFeatures? If not, should we check that as well?

Good catch, right now we aggregate processor features and tune features together in MCSubtargetInfo, so I don't think it makes sense to have a feature appears in both sets. I have updated the patch to print a warning for that situation as well.